src/dataset.ts

```diff
@@ -37,6 +37,16 @@ export interface Dataset {
      * environment. This may change in the future.
      */
     environments?: Environment[];
+    
+    /**
+     * Visualization settings to be used when loading the dataset.
+     * 
+     * Basically just a dictionary: visualization options will have to be 
+     * matched to whatever exists in a given implementation of the visualizer.
```

Comment:
This interface is used for the DefaultVisualizer only, so we can control what is accepted there. Instead of using an arbitrary object (`{ [name: string] : any };`), I would create an explicit interface for it and check that the dataset we get follow said interface. Something like 

```ts
interface MapSettings {
   color: string,
   x: string,
   y: string,
   z: string,
    // ...
}

presets?: {
    map?: Partial<MapSettings>,
    // ...
}
```

Reply:
Having an explicit interface allow to (a) better document what is accepted and (b) get the compiler to yell at us if we make a mistake.

Checking is allow to catch "wrong" input files early, and give the user a good error message.

Reply:
I created an interface for the input to the default visualizer, although it doesn't seem I ever need to use it...